### ISSUE TITLE
Ember Inspector not working on new / empty project

### ISSUE THREAD

**Issue by jacobq** [felixrieseberg/ember-electron#104]
Am I doing something wrong or is this a bug?
## Steps to reproduce
1. Run:

```
ember new example
cd example
ember install ember-electron # (overwrite: yes)
ember electron
```
1. Click "View" -> "Toggle Developer Tools"
2. Click "Ember" inspector tab
3. Get fishy Tomster error ("Ember application not detected!")
## Observations
- Command line output shows the following at start-up

```
Attempted to load extension "Ember Inspector" that has already been loaded.
```
- Chrome/Electron console shows

```
ember.debug.js:6797DEBUG: -------------------------------
ember.debug.js:6797DEBUG: Ember      : 2.8.2
ember.debug.js:6797DEBUG: Ember Data : 2.8.1
ember.debug.js:6797DEBUG: jQuery     : 3.1.1
ember.debug.js:6797DEBUG: -------------------------------
api.js:5 Installing Devtron from /home/jacob/tmp/example/node_modules/devtron
ember_debug.js:83 Ember Inspector Active
```
- Opening the Ember Inspector in an external browser appears to work
## System information

```
ember-cli: 2.8.0
node: 6.7.0
os: linux x64 (4.7.0)
```
## package.json

```
{
  "name": "example",
  "version": "0.0.0",
  "description": "Small description for example goes here",
  "private": true,
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "scripts": {
    "build": "ember build",
    "start": "ember server",
    "test": "ember test"
  },
  "repository": "",
  "engines": {
    "node": ">= 0.10.0"
  },
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "broccoli-asset-rev": "^2.4.2",
    "devtron": "1.4.0",
    "electron": "1.4.3",
    "electron-packager": "8.1.0",
    "electron-rebuild": "1.2.1",
    "ember-ajax": "^2.0.1",
    "ember-cli": "2.8.0",
    "ember-cli-app-version": "^1.0.0",
    "ember-cli-babel": "^5.1.6",
    "ember-cli-dependency-checker": "^1.2.0",
    "ember-cli-htmlbars": "^1.0.3",
    "ember-cli-htmlbars-inline-precompile": "^0.3.1",
    "ember-cli-inject-live-reload": "^1.4.0",
    "ember-cli-jshint": "^1.0.0",
    "ember-cli-qunit": "^2.1.0",
    "ember-cli-release": "^0.2.9",
    "ember-cli-sri": "^2.1.0",
    "ember-cli-test-loader": "^1.1.0",
    "ember-cli-uglify": "^1.2.0",
    "ember-data": "^2.8.0",
    "ember-electron": "1.10.7",
    "ember-export-application-global": "^1.0.5",
    "ember-inspector": "2.0.3",
    "ember-load-initializers": "^0.5.1",
    "ember-resolver": "^2.0.3",
    "ember-welcome-page": "^1.0.1",
    "loader.js": "^4.0.1"
  },
  "main": "electron.js",
  "ember-electron": {
    "WHAT IS THIS?": "Please see the README.md",
    "copy-files": [
      "electron.js",
      "package.json"
    ],
    "name": null,
    "platform": null,
    "arch": null,
    "version": null,
    "app-bundle-id": null,
    "app-category-type": null,
    "app-copyright": null,
    "app-version": null,
    "asar": null,
    "asar-unpack": null,
    "asar-unpack-dir": null,
    "build-version": null,
    "cache": null,
    "extend-info": null,
    "extra-resource": null,
    "helper-bundle-id": null,
    "icon": null,
    "ignore": null,
    "out": null,
    "osx-sign": {
      "identity": null,
      "entitlements": null,
      "entitlements-inherit": null
    },
    "overwrite": null,
    "prune": null,
    "strict-ssl": null,
    "version-string": {
      "CompanyName": null,
      "FileDescription": null,
      "OriginalFilename": null,
      "ProductName": null,
      "InternalName": null
    }
  }
}
```

**Comment by locks** [comment]
The problem here seems to be the same as https://github.com/ember-cli/ember-welcome-page/issues/22. In short, for a range of Ember CLI versions `ember new` doesn't generate an application.hbs. The welcome page is served via an express.js middleware, so there's no actual Ember application for the Inspector to register.
This is fixed in master.

**Comment by mklnz** [comment]
I'm having a similar but slightly different issue with ember inspector. It appears it works inside a different browser window pointing to e.g. http://localhost:30820/, the ember inspector in Chrome detects the application just fine. However, if I open the ember inspector inside the actual electron window, it says ember application not found. What could be the cause of this?

**Comment by felixrieseberg** [comment]
☝️  Hm, that's fascinating. Is there *any* way for you to get Ember Inspector to detect your app in Electron?

**Comment by mklnz** [comment]
Just an update, I still haven't figured out how to get Ember inspector working. I suspect it is due to file:// access, even though it is already enabled in Chrome itself.

Upon further testing, by opening Chrome itself and pointing to the dist/index.html file using file://, Ember inspector does not detect the application. Perhaps this is an issue with the Ember inspector addon itself?

**Comment by anulman** [comment]
@mklnz can you check if this is resolved by @pichfl's fix in #139 ?

**Comment by pichfl** [comment]
I think this might be more related to #140 which might actually be a duplicate of this one here.

**Comment by jacobq** [comment]
I gave this a quick whirl this morning. Following the steps listed at the beginning still fails (i.e. using stable/release branches). I'll try again with master now and post a reply if it's different.

## Output from stable/release attempt (failed)
```
ember new example
installing app
  create .bowerrc
  create .editorconfig
  create .ember-cli
  create .jshintrc
  create .travis.yml
  create .watchmanconfig
  create README.md
  create app/app.js
  create app/components/.gitkeep
  create app/controllers/.gitkeep
  create app/helpers/.gitkeep
  create app/index.html
  create app/models/.gitkeep
  create app/resolver.js
  create app/router.js
  create app/routes/.gitkeep
  create app/styles/app.css
  create app/templates/application.hbs
  create app/templates/components/.gitkeep
  create bower.json
  create config/environment.js
  create ember-cli-build.js
  create .gitignore
  create package.json
  create public/crossdomain.xml
  create public/robots.txt
  create testem.js
  create tests/.jshintrc
  create tests/helpers/destroy-app.js
  create tests/helpers/module-for-acceptance.js
  create tests/helpers/resolver.js
  create tests/helpers/start-app.js
  create tests/index.html
  create tests/integration/.gitkeep
  create tests/test-helper.js
  create tests/unit/.gitkeep
  create vendor/.gitkeep
Successfully initialized git.
Installed packages for tooling via npm.
Installed browser packages via Bower.
$ cd example
$ ember install ember-electron
Installed packages for tooling via npm.
installing ember-electron
? Overwrite config/environment.js? Yes, overwrite
  overwrite config/environment.js
  create electron.js
  create tests/electron.js
  create tests/package.json
  overwrite package.json
Ember Electron requires configuration. Please consult the Readme to ensure that this addon works!
https://github.com/felixrieseberg/ember-electron
  install packages electron, electron-packager, electron-rebuild, ember-inspector, devtron
Installed packages for tooling via npm.
Installed addon package.
$ ember electron

Build successful - 4929ms.



Slowest Nodes (totalTime => 5% )              | Total (avg)         
----------------------------------------------+---------------------
Babel (16)                                    | 4590ms (286 ms)     

--------------------------------------------------------------------
Ember Inspector running on http://localhost:30820
Open the inspector URL in a browser to debug the app!
The inspector is also available as an addon inside your
app's developer tools.
--------------------------------------------------------------------
Starting Electron...
--------------------------------------------------------------------
Reading tmp/example/node_modules/devtron/manifest.json failed.
Error: ENOENT: no such file or directory, open 'tmp/example/node_modules/devtron/manifest.json'
    at Error (native)
    at Object.fs.openSync (fs.js:640:18)
    at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:173:20)
    at Object.fs.readFileSync (fs.js:508:33)
    at Object.fs.readFileSync (ELECTRON_ASAR.js:506:29)
    at getManifestFromPath (temp/example/node_modules/electron/dist/resources/electron.asar/browser/chrome-extension.js:34:26)
    at App.<anonymous> (temp/example/node_modules/electron/dist/resources/electron.asar/browser/chrome-extension.js:363:26)
    at App.g (events.js:286:16)
    at emitTwo (events.js:111:20)
    at App.emit (events.js:191:7)
[31939:0223/075618:ERROR:CONSOLE(6954)] "SyntaxError: Unexpected token ( in JSON at position 0", source: chrome-devtools://devtools/bundled/inspector.js (6954)
```

```
Failed to parse SourceMap: http://localhost:30820/socket.io/socket.io.js.map
ember.debug.js:5720DEBUG: -------------------------------
ember.debug.js:5720DEBUG: Ember      : 2.11.2
ember.debug.js:5720DEBUG: Ember Data : 2.11.1
ember.debug.js:5720DEBUG: jQuery     : 3.1.1
ember.debug.js:5720DEBUG: -------------------------------
api.js:5 Installing Devtron from /home/jquant/Documents/KFMT/Software/temp/example/node_modules/devtron
ember_debug.js:83 Ember Inspector Active
/ember-welcome-page/images/construction.png:1 GET file:///ember-welcome-page/images/construction.png net::ERR_FILE_NOT_FOUND
```

**Comment by anulman** [comment]
Seems you're trying to load ember-welcome-page. Have you tried rm'ing it, per @locks' note? https://github.com/felixrieseberg/ember-electron/issues/104#issuecomment-257140164

**Comment by locks** [comment]
@anulman that problem should be fixed in 2.11 though, `ember-welcome-page` is no longer express middleware, we add a component to `application.hbs`.

**Comment by anulman** [comment]
@locks I see, thanks for the update! Imagine this issue has two pieces to it then:

1) @jacobq are you having issues with how welcome-page is rendering? That last line of browser console output + [this line in welcome-page](https://github.com/ember-cli/ember-welcome-page/blob/master/addon/templates/components/welcome-page.hbs#L4) seem to indicate a relative URL for the png, which is not going to work on a fs (vs. network) URL lookup.

2) Inspector issues, which are likely related to #140

**Comment by anulman** [comment]
@jacobq was reviewing this with @pichfl earlier today; can you confirm whether the Inspector issues were with:

- The remote inspector (i.e. visiting `http://localhost:30820` in your browser); vs.
- The embedded inspector (i.e. opening up devtools from your Electron app window & checking the Ember tab)?

**Comment by jacobq** [comment]
@anulman My issue is unrelated to `ember-welcome-page`. What I want to be able to do is use the Ember Inspector from the dev tools of the actual electron app I am developing. When I found that it was not working I tried to use it with an empty / new project and realized that didn't work either. That's when I created this GH issue. 

Using `http://localhost:30820` to inspect the application from another browser works OK (though it often needs to be refreshed to work right). It's just the embedded one that I am having trouble with.

Here's a screenshot of the problem using ember-electron v1.12.7. I will try with the latest commits on master to see if that fixes it.

![sample-error](https://cloud.githubusercontent.com/assets/1633459/23519155/8c4647b8-ff3b-11e6-9e8e-b09f5725460c.png)

**Comment by jacobq** [comment]
I just tried running
```
ember new example
cd example
ember install https://github.com/felixrieseberg/ember-electron.git#master
```
I ignored this: "WARNING: Could not figure out blueprint name from: "https://github.com/felixrieseberg/ember-electron.git". Please install the addon blueprint via "ember generate <addon-name>" if necessary." then ran `nombom; ember electron` and encountered the following error:

```
You must depend on "electron-prebuilt-compile" in your devDependencies
```

So I ran `npm install --save-dev electron-prebuilt-compile@1.6.1`, which seemed to complete successfully, but then running `ember electron` resulted in an error: `You must depend on an EXACT version of "electron-prebuilt-compile" not a range"` so I edited `package.json` and removed the preceding `^` from the version string then ran `ember electron` again. This time I get `Failed to locate startable Electron application`, and I'm guessing this has to do with the blueprint not running correctly.

If someone will help me understand how to test this properly I will try again. Otherwise I can wait until the next release.

**Comment by anulman** [comment]
Awesome, thanks so much for these details @jacobq.

Re: embedded inspector, this was fixed by #148, and should be available in the 2.0 beta you're trying to install.

Re: electron-prebuilt-compile errors, you're going to need to take a couple of extra steps while we wait for a few upstream PRs to land. The install steps are listed in the *FYI* section in #147's first message.

Closing this issue since A) the inspector issue you've reported is known for this version + resolved for later ones; and B) any upgrade process discussion should take place in #147
